extension-manifest-v2/src/background.js

```diff
@@ -552,6 +556,30 @@ function onMessageHandler(request, sender, callback) {
 	if (origin === 'blocked_redirect') {
 		return handleBlockedRedirect(name, message, tab_id, callback);
 	}
+	if (origin === 'autoconsent') {
+		if (name === 'enable') {
+			conf.enable_autoconsent = true;
+			if (message.url) {
+				conf.autoconsent_whitelist = conf.autoconsent_whitelist.concat(message.url);
+			} else {
+				conf.autoconsent_whitelist = [];
+				conf.autoconsent_blacklist = [];
+			}
+
+			return false;
+		}
+		if (name === 'disable') {
+			if (message.url) {
+				conf.autoconsent_blacklist = conf.autoconsent_blacklist.concat(message.url);
+			} else {
+				conf.enable_autoconsent = false;
+				conf.autoconsent_whitelist = [''];
```

Comment:
is this correct to have an empty string here?

Reply:
This is a trick to differ from the state where the user choose his preference for all or selected websites. However, I think it might be confusing so that I will change that to `null` or an array with a list of domains.